src/extension.ts

```diff
@@ -863,6 +865,84 @@ export function activate(context: vscode.ExtensionContext) {
     )
   );
 
+  subscriptions.push(
+    vscode.commands.registerCommand("vscode-couchbase.openSampleProjects", async () => {
+      try {
+        const panel = vscode.window.createWebviewPanel(
+          "sampleProjects",
+          "Sample Projects",
+          vscode.ViewColumn.One,
+          {
+            enableScripts: true,
+          }
+        );
+        panel.webview.html = getSampleProjects();
+        panel.webview.onDidReceiveMessage(
+          async (message) => {
+            if (!message || !message.repo) {
+              return;
+            }
+
+            const projectName = await vscode.window.showInputBox({
+              prompt: "Project Name",
+              placeHolder: "Project-Name",
+              ignoreFocusOut: true,
+              value: "",
+            });
+            if (!projectName) {
+              vscode.window.showErrorMessage("Project name is required.");
+              return;
+            }
+
+            const fileUris = await vscode.window.showOpenDialog({
+              openLabel: "Select location",
+              canSelectMany: false,
+              canSelectFiles: false,
+              canSelectFolders: true,
+            });
+            if (!fileUris || !fileUris[0]) {
+              vscode.window.showErrorMessage("Project location is required.");
+              return;
+            }
+
+            const projectUri = vscode.Uri.joinPath(fileUris[0], projectName);
+            const repoUrl = `couchbase-examples/${message.repo}`;
+
+            await gitly(repoUrl, projectUri.path, { throw: true });
+
+            vscode.window.showInformationMessage(``);
+            let answer = await vscode.window.showInformationMessage(
+              `Example project created at: ${projectUri.path}.`,
+              ...["Open", "Add to Workspace"]
+            );
+            switch (answer) {
+              case "Open":
+                await vscode.commands.executeCommand(
+                  "vscode.openFolder",
+                  projectUri
+                );
+                break;
+              case "Add to Workspace":
+                vscode.workspace.updateWorkspaceFolders(
+                  vscode.workspace.workspaceFolders
+                    ? vscode.workspace.workspaceFolders.length
+                    : 0,
+                  null,
+                  { uri: projectUri }
+                );
```

Comment:
Are we suppose to close the sampleProject panel after adding project to the workspace
If yes please close it
If no, then here also we are facing the same error as https://github.com/couchbaselabs/VSCode-Couchbase/pull/208#issuecomment-1549263921

Reply:
Thanks @AayushTyagi1 - added a commit to close the panel after a sample has been crated in [1b70e9e](https://github.com/couchbaselabs/VSCode-Couchbase/pull/208/commits/1b70e9e37fe131bdcd16d251cad456a02414924c)